python/ray/tune/function_runner.py

```diff
@@ -476,34 +477,37 @@ def detect_checkpoint_function(train_func, abort=False):
     return validated
 
 
-def wrap_function(train_func):
+def wrap_function(train_func, warn=True):
     if hasattr(train_func, "__mixins__"):
         inherit_from = train_func.__mixins__ + (FunctionRunner, )
     else:
         inherit_from = (FunctionRunner, )
 
+    func_args = inspect.getfullargspec(train_func).args
+    use_checkpoint = detect_checkpoint_function(train_func)
+    if len(func_args) > 1:  # more arguments than just the config
+        if "reporter" not in func_args and not use_checkpoint:
```

Comment:
Right now this doesnt catch the case of `def train(config, xxx, reporter)`.

Reply:
Do you mean that we will try to call `train()` without filling the `xxx` variable?

Reply:
I guess this was the case before, too, but if that's a bug we should fix it.